azure-iot-device/azure/iot/device/common/transport/mqtt/pipeline_stages.py

```diff
@@ -0,0 +1,645 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import abc
+import six
+from collections import deque
+from . import pipeline_ops
+from . import pipeline_events
+from . import mqtt_topic_helper
+from azure.iot.device.iothub.models import Message
+from azure.iot.device.common.transport.mqtt.mqtt_provider import MQTTProvider
+from azure.iot.device.common.transport import constant
+
+logger = logging.getLogger(__name__)
+
+# TODO: standardize on {} instead of %s
+# TODO: standardize logs with {}({}) prefix
+
+
+@six.add_metaclass(abc.ABCMeta)
+class PipelineStage(object):
```

Comment:
Given that (some of) these are high level, protocol-agnostic concepts, does it make sense for them to all be in transport.mqtt? Shouldn't some of these be in transport rather than transport.mqtt?

Reply:
yes, absolutely.  splitting these files is one of my next steps


---
In reply to: [279665186](https://github.com/Azure/azure-iot-sdk-python-preview/pull/87#discussion_r279665186) [](ancestors = 279665186)